artiq/firmware/ddb_parser/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use thiserror::Error;
+
+use indoc::indoc;
+use pyo3::types::{PyDict, PyModule};
+use std::collections::HashMap;
+
+mod devices;
+pub use devices::Device;
+
+pub type DeviceDb = HashMap<String, Device>;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("type error: {0}")]
+    Type(String),
+    #[error(transparent)]
+    Python(#[from] pyo3::PyErr),
+    #[error(transparent)]
+    Parse(#[from] serde_json::Error),
+}
+
+pub fn parse(python_code: &str) -> Result<DeviceDb, Error> {
+    pyo3::Python::with_gil(|py| {
+        let module = PyModule::from_code(py, python_code, "device_db.py", "device_db")?;
+        let ddb = module
```

Comment:
It is a little bit confusing that the name is reused later. Maybe:

```suggestion
        let ddb_dict = module
```

Reply:
Addressed in [17635bb](https://github.com/alpine-quantum-technologies/artiq/pull/5/commits/17635bb10acdf99795316b9ab09c19b3a82513d5).